lib/github-pages-health-check.rb

```diff
@@ -65,11 +66,15 @@ def valid_domain?
       PublicSuffix.valid? domain
     end
 
-    # Is this domain an SLD, meaning a CNAME would be innapropriate
+    # Is this domain an apex domain, meaning a CNAME would be innapropriate
     def apex_domain?
-      PublicSuffix.parse(domain).trd == nil
-    rescue
-      false
+      return @apex_domain if defined?(@apex_domain)
+
+      answers = Resolv::DNS.open { |dns|
+        dns.getresources(absolute_domain, Resolv::DNS::Resource::IN::NS)
```

Comment:
I went with Resolv because it was acting up on one of the test cases and returning records it shouldn't have:

```
[5] pry(main)> Net::DNS::Resolver.start("blog.parkermoore.de", Net::DNS::NS).answer
=> [blog.parkermoore.de.    261     IN      CNAME   parkr.github.io., parkr.github.io.        3561    IN      CNAME   github.map.fastly.net.]
[6] pry(main)> Resolv::DNS.open { |dns| dns.getresources("blog.parkermoore.de", Resolv::DNS::Resource::IN::NS) }
=> []
```

But we can probably just filter non-NS records out in Rubyland